lambdas/transcode/transcode/index.py

```diff
@@ -0,0 +1,51 @@
+"""
+Generate video previews for videos in S3.
+"""
+import pathlib
+import subprocess
+import tempfile
+
+from t4_lambda_shared.decorator import api, validate
+from t4_lambda_shared.utils import get_default_origins
+
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'url': {
+            'type': 'string'
+        }
+    },
+    'required': ['url'],
+    'additionalProperties': False
+}
+
+FFMPEG = pathlib.Path('/opt/bin/ffmpeg')
+FFMPEG_TIME_LIMIT = 20
+DURATION = 5
+WIDTH = 320
+HEIGHT = 240
+
+
+@api(cors_origins=get_default_origins())
+@validate(SCHEMA)
+def lambda_handler(request):
```

Comment:
We should have one unit test that calls this handler (all other lambdas have one). It can be simple / use tiny files. That way it gets invoked under 3.6, 3.7, 3.8, etc.

Reply:
And then we'll need to add this lambda to the CircleCI pass.

Reply:
Since this depends on a binary, for unit test you can shim out the subprocess call (still useful test imo; b/c the calling code runs), or (not necessary) do what I did for thumbnail and make some tests conditional on a flag.